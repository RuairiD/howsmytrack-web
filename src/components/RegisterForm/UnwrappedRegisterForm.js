import React from "react";

import { Button, Icon, Input, Form } from "antd";

const UnwrappedRegisterForm = ({
    onSubmit,
    form,
    disabled,
}) => (
    <Form onSubmit={onSubmit} className="hmt-form">
        <Form.Item>
            {form.getFieldDecorator("email", {
                rules: [{ required: true, message: "Please enter your email address." }],
            })(
                <Input
                    prefix={<Icon type="user" style={{ color: "rgba(0,0,0,.25)" }} />}
                    placeholder="Email"
                    autoFocus
                />,
            )}
        </Form.Item>
        <Form.Item>
            {form.getFieldDecorator("password", {
                rules: [{ required: true, message: "Please enter your password." }],
            })(
                <Input
                    prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
                    type="password"
                    placeholder="Password"
                />,
            )}
        </Form.Item>
        <Form.Item>
            {form.getFieldDecorator("passwordRepeat", {
                rules: [{ required: true, message: "Please repeat your password." }],
            })(
                <Input
                    prefix={<Icon type="lock" style={{ color: "rgba(0,0,0,.25)" }} />}
                    type="password"
                    placeholder="Confirm Password"
                />,
            )}
        </Form.Item>
        <Form.Item>
            <Button
                block
                type="primary"
                htmlType="submit"
                disabled={disabled}
            >
                Register
            </Button>
        </Form.Item>
    </Form>
);

export default UnwrappedRegisterForm;
