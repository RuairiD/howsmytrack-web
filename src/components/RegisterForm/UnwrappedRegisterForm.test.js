import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import UnwrappedRegisterForm from "./UnwrappedRegisterForm";

jest.mock("../../apiRoot", () => "http://localhost:8000");

const mockValidForm = {
    getFieldDecorator: jest.fn(() => (c) => c),
    validateFieldsAndScroll: jest.fn((afterValidation) => afterValidation(false, {
        success: true,
        error: null,
    })),
};

describe("UnwrappedRegisterForm", () => {
    it("renders a blank form for registering a new account", () => {
        const wrapper = shallow(
            <UnwrappedRegisterForm
                form={mockValidForm}
                onSubmit={() => {}}
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("disables Submit button if disabled is set (e.g. if request is loading or form has already been submitted)", () => {
        const wrapper = shallow(
            <UnwrappedRegisterForm
                form={mockValidForm}
                onSubmit={() => {}}
                disabled
            />,
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
