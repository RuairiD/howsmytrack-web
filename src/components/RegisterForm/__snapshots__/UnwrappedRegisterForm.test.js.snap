// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnwrappedRegisterForm disables Submit button if disabled is set (e.g. if request is loading or form has already been submitted) 1`] = `
<Form
  className="hmt-form"
  colon={true}
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
>
  <FormItem
    hasFeedback={false}
  >
    <Input
      autoFocus={true}
      placeholder="Email"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="user"
        />
      }
      type="text"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Password"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Confirm Password"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Button
      block={true}
      disabled={true}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Register
    </Button>
  </FormItem>
</Form>
`;

exports[`UnwrappedRegisterForm renders a blank form for registering a new account 1`] = `
<Form
  className="hmt-form"
  colon={true}
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
>
  <FormItem
    hasFeedback={false}
  >
    <Input
      autoFocus={true}
      placeholder="Email"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="user"
        />
      }
      type="text"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Password"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Input
      placeholder="Confirm Password"
      prefix={
        <Icon
          style={
            Object {
              "color": "rgba(0,0,0,.25)",
            }
          }
          type="lock"
        />
      }
      type="password"
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
  >
    <Button
      block={true}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Register
    </Button>
  </FormItem>
</Form>
`;
